sdk/eventhub/eventhubs-checkpointstore-table/test/tables-checkpointstore.spec.ts

```diff
@@ -0,0 +1,527 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import chai from "chai";
+const should = chai.should();
+import { TableCheckpointStore } from "../src";
+import debugModule from "debug";
+const debug = debugModule("azure:event-hubs:tableCheckpointStore");
+import { Checkpoint, PartitionOwnership } from "@azure/event-hubs";
+import { EnvVarKeys, getEnvVars } from "./utils/testUtils";
+import { TableServiceClient, AzureNamedKeyCredential, TableClient } from "@azure/data-tables";
+import { CheckpointEntity, PartitionOwnershipEntity } from "../src/tableCheckpointStore";
+
+const env = getEnvVars();
+
+/* test to show that test framework is set up well */
+describe("TableCheckpointStore", () => {
+  it("is exported from the package", () => {
+    should.exist(TableCheckpointStore, "Expected TableCheckpointStore to be exported.");
+  });
+});
+
+const service = {
+  storageAccountName: env[EnvVarKeys.STORAGE_ACCOUNT_NAME],
+  storageAccountKey: env[EnvVarKeys.STORAGE_ACCOUNT_KEY]
+};
+
+const credential = new AzureNamedKeyCredential(
+  service.storageAccountName,
+  service.storageAccountKey
+);
+const serviceClient = new TableServiceClient(
+  `https://${service.storageAccountName}.table.core.windows.net`,
+  credential
+);
+
+describe("TableCheckpointStore", function(): void {
+  let client: TableClient;
+  let tableName: string;
+
+  describe("Runs tests on table with no entities", function() {
+    beforeEach("creating table", async () => {
+      tableName = `table${new Date().getTime()}${Math.floor(Math.random() * 10) + 1}`;
+      client = new TableClient(
+        `https://${service.storageAccountName}.table.core.windows.net`,
+        tableName,
+        credential
+      );
+      await serviceClient.createTable(tableName);
+    });
+    afterEach(async () => {
+      await serviceClient.deleteTable(tableName);
+    });
```

Comment:
The beforeEach and afterEach could be moved to be above your describe call. Then you wouldn't need to duplicate table creation/deletion in your tests that do have entities.

Reply:
yes